googleapiclient/discovery_cache/file_cache.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2014 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""File based cache for the discovery document.
+The cache is stored in a single file so that multiple processes can
+
+share the same cache. It locks the file whenever accesing to the
```

Comment:
Okay, but I'm still confused by the use of tempfile. How would it be that two processes would share a single cache? I can see how one process creates a cache file for its own use, but how would it communicate the name of the file to be used to an additional process?